tracer/build/_build/Build.GitHub.cs

```diff
@@ -94,6 +99,92 @@ await client.Issue.Update(
             Console.WriteLine($"PR assigned");
         });
 
+    Target LLMReport => _ => _
+           .Unlisted()
+           .Requires(() => GitHubRepositoryName)
+           .Requires(() => GitHubToken)
+           .Requires(() => OpenAIKey)
+           .Requires(() => PullRequestNumber)
+           .Executes(async () =>
+           {
+               await GenerateLLMReport(false);
+           });
+
+    Target LLMLocalReport => _ => _
+       .Unlisted()
+       .Requires(() => GitHubRepositoryName)
+       .Requires(() => GitHubToken)
+       .Requires(() => OpenAIKey)
+       .Requires(() => PullRequestNumber)
+       .Executes(async () =>
+       {
+           await GenerateLLMReport(true);
+       });
+
+    private async Task GenerateLLMReport(bool executeLocal = true)
+    {
+        var excludePath = new string[] { "Datadog.Trace/Generated", ".g.cs" };
+        var extensionsToReview = new[] { ".csproj", ".cs", ".yml", ".h", ".cpp", ".dockerfile" };
+        var prompt = "Review this pull request with a focus on improving performance and bug detection. Make a list of the most important areas that need enhancement. For each suggestion, name the involved file and include both the original code and your recommended change, adding the corrected code if applicable. The code to be reviewed is the result of running the \"git --diff\" command. Highlight any performance bottlenecks and opportunities for optimization." + Environment.NewLine;
+
+        string result = string.Empty;
+        var client = GetGitHubClient();
+
+        var pullRequest = await client.PullRequest.Get(GitHubRepositoryOwner, GitHubRepositoryName, PullRequestNumber.Value);
+        var pullRequestFiles = await client.PullRequest.Files(GitHubRepositoryOwner, GitHubRepositoryName, PullRequestNumber.Value);
+
+        string changesText = string.Empty;
+        foreach (var file in pullRequestFiles)
+        {
+            if (!extensionsToReview.Any(ext => file.FileName.EndsWith(ext)) || excludePath.Any(x => file.FileName.Contains(x, StringComparison.OrdinalIgnoreCase)))
+            {
+                continue;
+            }
+
+            changesText += ($"Filename: {file.FileName}" + Environment.NewLine + ($"{file.Patch}") + Environment.NewLine + Environment.NewLine);
+        }
+
+        if (string.IsNullOrEmpty(changesText))
+        {
+            result = "No changes detected.";
+        }
+        else if (string.IsNullOrEmpty(OpenAIKey))
+        {
+            result = "Null or empty OpenAI key.";
+        }
+        else
+        {
+            var fullPrompt = prompt + changesText;
+
+            result = OpenAiApiCall.TryGetReponse(ref fullPrompt, OpenAIKey);
+
+            if (executeLocal)
+            {
+                File.WriteAllText("changes.txt", fullPrompt);
```

Comment:
I think you should make this absolute paths, so the location of `changes.txt` is fixed and in a sensible (i.e. temporary) place. That way it will be excluded by the gitignore (same goes for the `LLMResult.txt`).

Personally, I'd rather we didn't generate these files, and instead just print them to the console. You can always pipe them to a file if you want to anyway.

Reply:
Agree. Thanks! Done!